File: leibnizcomputepi.py
true = True
false = False
def pi(maxnum):
    piover4 = 0
    negative = false
    for n in range (1, maxnum, 2):
        if(negative):
            piover4 -= 1 / n
        else:
            piover4 += 1 / n
        negative = not negative
    print(piover4 * 4)

File: leibnizcomputepi.cpp
#include <iostream>
#include <string>

#define debuggin

using namespace std;

int main(int argc, char** argv)
{    
    double piover4 = 0;
    
    uint64_t maxnum = stoll(argv[1]);
    
    bool negative = false;
    for(uint64_t n = 1; n < maxnum; n += 2)
    {
        if(negative)
        {
            #ifdef debugging
            cout << -1 / (double)n << endl;
            #endif
            piover4 -= (1 / (double)n);
        }
        else
        {
            #ifdef debugging
            cout << 1 / (double)n << endl;
            #endif
            piover4 += (1 / (double)n);
        }
        
        negative = !negative;
    }

    cout << piover4 * 4 << endl;
    
    return 0;
}

